flink-core/src/main/java/org/apache/flink/configuration/NettyShuffleEnvironmentOptions.java

```diff
@@ -175,7 +175,8 @@
 
 	public static final ConfigOption<Integer> NUM_ARENAS =
 		key("taskmanager.network.netty.num-arenas")
-			.defaultValue(-1)
+			.intType()
```

Comment:
Why do we need this line? Isn't this option already declared as `ConfigOption<Integer>`?

Reply:
this adds an option class explicitly, `ConfigOption<Integer>` is typed only during compilation. From what I see, `defaultValue()` for an untyped option has been deprecated. The details in FLIP-77.